query_geospatial_test.go

```diff
@@ -1,42 +1,89 @@
 package gorethink
 
 import (
+	"reflect"
+
 	"github.com/dancannon/gorethink/types"
 
 	test "gopkg.in/check.v1"
 )
 
+/* BEGIN FLOAT HELPERS */
+
+// totally ripped off from math/all_test.go
+// https://github.com/golang/go/blob/master/src/math/all_test.go#L1723-L1749
+func tolerance(a, b, e float64) bool {
+	d := a - b
+	if d < 0 {
+		d = -d
+
+	}
+
+	if a != 0 {
+		e = e * a
+		if e < 0 {
+			e = -e
+
+		}
+
+	}
+	return d < e
+}
+
+func mehclose(a, b float64) bool    { return tolerance(a, b, 1e-2) }
+func kindaclose(a, b float64) bool  { return tolerance(a, b, 1e-8) }
+func prettyclose(a, b float64) bool { return tolerance(a, b, 1e-14) }
+func veryclose(a, b float64) bool   { return tolerance(a, b, 4e-16) }
+func soclose(a, b, e float64) bool  { return tolerance(a, b, e) }
+
+func compareCoordinates(co [][]float64, lines types.Lines, c *test.C) {
+	for _, points := range lines {
+		for ip, point := range points {
+			v := reflect.ValueOf(point)
+			for i := 0; i < v.NumField(); i++ {
```

Comment:
Hah. Much simpler. But in that case I'd suggest the following for readability and error reporting.
 
```go
func compareCoordinates(co [][]float64, lines types.Lines, c *test.C) {
	for _, points := range lines {
		for ip, point := range points {
			lon := co[ip][0]
			lat := co[ip][1]
			if !kindaclose(lon, point.Lon) {
				c.Errorf("the deviation between the compared floats is too great [%v:%v]", lon, point.Lon)
			}
			if !kindaclose(lat, point.Lat) {
				c.Errorf("the deviation between the compared floats is too great [%v:%v]", lat, point.Lat)
			}
		}
	}
}
```